HomeAssistant.py

```diff
@@ -1048,22 +1045,29 @@ def mergeDialogSlots(self):
 						activeDialogFile['slotTypes'][x]['values'] = backupUserSlots
 						filePath.write_text(json.dumps(activeDialogFile, ensure_ascii=False, indent=4))
 
-	#Merge dialogTemplate files on Update if a backup exists
+		self.logInfo(f'Merged DialogTemplate files. You "might" need to re train Alice')
+
+
+	# Merge dialogTemplate files on Update if a backup exists and restore My home display settings
 	def onSkillUpdated(self, skill: str):
 		if skill in self.name and self.getConfig('enableBackup'):
-			dialogFile = Path(self.getResource(f'BackUp/{self.activeLanguage()}.json'))
+			dialogFile = self.getResource(f'Backup/{self.activeLanguage()}.json')
 
 			if dialogFile.exists():
-				self.mergeDialogIntents()
-				self.logInfo(f'Merged DialogTemplate files. You may need to re train Alice')
+				self.restoreBackUpFiles()
+				self.say(
+					text=self.randomTalk(text='restored')
+				)
 
-	#onStop. backup display coordinates and dialogTemplate file
+
+	# onStop. backup display coordinates and dialogTemplate file
 	def onStop(self):
 		if self.getConfig('enableBackup'):
-			self.runBackup() #save dialogTemplate file and display settings to backup directory
+			self.runBackup()  # save dialogTemplate file and display settings to backup directory
 		else:
 			self.logInfo(f'![green](Stopped)')
```

Comment:
def onStop(self):
    if self.getConfig('enableBackup'):
        self.runBackup()
    super.onStop()

Reply:
If you don't super call the onStop, the skill is not correctly stopped, intents are not deactivated, skill still appears as active etc etc

Reply:
interesting, That would explain why mpbs still had debug messages after uninstalling the skill perhaps.

adding that to pycharm comes back with "unresolved attribute reference to 'onStop' for class 'super'

Reply:
Should not: https://puu.sh/GwUc3.png

Maybe you haven't configured the project dependency in pycharm?

Reply:
ohh crap...
```HomeAssistant]           Stopped
Error while shutting down manager SkillManager: type object 'super' has no attribute 'onStop'
Traceback (most recent call last):
  File "/home/pi/ProjectAlice/core/base/SuperManager.py", line 206, in onStop
    manager.onStop()
  File "/home/pi/ProjectAlice/core/base/SkillManager.py", line 398, in onStop
    skillItem.onStop()
  File "/home/pi/ProjectAlice/skills/HomeAssistant/HomeAssistant.py", line 1067, in onStop
    super.onStop()
AttributeError: type object 'super' has no attribute 'onStop'
```

Reply:
WTF? Can't be, HomeAssistant class is an AliceSkill

See the o with up arrow? That means it calls the superclass
https://puu.sh/GwUE9.png

Reply:
ahh you have super() in that example. that might be the reason , was just super.onStop() previously : )

Reply:
Lol, must be dislexic today, did not even pay attention to it :)